nest/core/protocols.py

```diff
@@ -0,0 +1,464 @@
+# protocols.py
+
+from __future__ import annotations
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    List,
+    Optional,
+    Protocol,
+    Type,
+    TypeVar,
+    Union,
+    runtime_checkable,
+    Generic,
+)
+
+###############################################################################
+# 1. REQUEST & RESPONSE
+###############################################################################
+
+@runtime_checkable
+class RequestProtocol(Protocol):
+    """
+    Abstract representation of an HTTP request.
+    """
+
+    @property
+    def method(self) -> str:
```

Comment:
should change type to Literal['GET' | 'POST' | 'PUT' | 'DELETE']
Or string eum, Thus allowing  string manipulation and string input